test/e2e/core.go

```diff
@@ -0,0 +1,71 @@
+package e2e
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net"
+	"strings"
+)
+
+type command struct {
+	cmd       string
+	component string
+}
+
+func coreDump(dir string) {
+	c, err := loadClient()
+	if err != nil {
+		fmt.Printf("Error creating client: %v", err)
+		return
+	}
+	provider := testContext.Provider
+	if !providerIs("gce", "gke") {
+		fmt.Printf("Skipping SSH core dump, which is not implemented for %s", provider)
+		return
+	}
+
+	// Get all nodes' external IPs.
+	hosts, err := NodeSSHHosts(c)
+	if err != nil {
+		fmt.Printf("Error getting node hostnames: %v", err)
+		return
+	}
+
+	cmds := []command{
+		{"cat /var/log/kubelet.log", "kubelet"},
+		{"cat /var/log/kube-proxy.log", "kube-proxy"},
+	}
+	logCore(cmds, hosts, dir, provider)
+	
+	// I wish there was a better way to get the master IP...
+	config, err := loadConfig()
+	if err != nil {
+		fmt.Printf("Error loading config: %v")
+	}
+	ix := strings.LastIndex(config.Host, "/")
+	master := net.JoinHostPort(config.Host[ix + 1:], "22")
+	cmds = []command{
+		{"cat /var/log/kube-apiserver.log", "kube-apiserver"},
+		{"cat /var/log/kube-scheduler.log", "kube-scheduler"},
+		{"cat /var/log/kube-controller-manager.log", "kube-controller-manager"},
+	}
+	logCore(cmds, []string{master}, dir, provider)
+}
+
+func logCore(cmds []command, hosts []string, dir, provider string) {
+	// Run commands on all nodes via SSH.
+	for _, cmd := range cmds {
+		fmt.Printf("SSH'ing to all nodes and running %s\n", cmd.cmd)
+		for _, host := range hosts {
+			logfile := fmt.Sprintf("%s/%s-%s.log", dir, host, cmd.component)
+			fmt.Printf("Writing to %s.\n", logfile)
+			stdout, stderr, _, err := SSH(cmd.cmd, host, provider)
```

Comment:
It would be super duper to run these in separate go routines to speed up for large clusters (e.g. we have 100-node ones for scalability tests).